features/lwipstack/lwip/src/core/ipv4/lwip_ip4_frag.c

```diff
@@ -662,8 +662,8 @@ ip4_reass(struct pbuf *p)
     }
 
     /* release the sources allocate for the fragment queue entry */
-    if (ipr_prev != NULL) {
-      ip_reass_dequeue_datagram(ipr, ipr_prev);
+    if (reassdatagrams != NULL) {
```

Comment:
No, this isn't right. Correct fix, afaict is to just remove the `if`.

We want to free the entry we just reassembled, `ipr`, and the free routine needs to know the previous entry in the list, if it's not the first (ie `==reassdatagrams`).

Looks like it was broken by #10647 - a "fix" for a false positive from Coverity?

Reply:
Of course, it may have been a valid Coverity report, and a real fix is still required, rather than just reverting.

Reply:
Also note that #10647 changed two instances of this code, so there are two places potentially needing modification.